packages/core/strapi/bin/strapi.js

```diff
@@ -255,4 +257,157 @@ program
   .action(getLocalScript('ts/generate-types'));
 
+// `$ strapi export`
+const parseBool = (arg) => {
+  try {
+    return parseType({ type: 'boolean', value: arg });
+  } catch (e) {
+    return false;
+  }
+};
+// // Will be used for the options that accept a list
+// const listOption = (value) => {
+//   return value.split(',');
+// };
+
+program
+  .command('export')
+  .description('Export data from Strapi to file')
+  .option(
+    '--output <outputFilename>',
+    'Filename to output (without extension) [Default: "export_{yyyymmddHHMMSS}[.tar.gz][.gpg]"]'
+  )
+  .addOption(
+    new Option('--encrypt [boolean]', 'encrypt output file').default(true).argParser(parseBool)
+  )
+  .addOption(new Option('--no-encrypt', 'do not encrypt output file'))
+  .addOption(
+    new Option('--compress [boolean]', 'compress content').default(true).argParser(parseBool)
+  )
+  .addOption(new Option('--no-compress', 'do not compress content'))
+  .addOption(new Option('--key', 'provide encryption key in command instead of using a prompt'))
+  // Options we plan to add in the future:
+  // .option('--config <configFile>', 'Path to the config file')
+  // .addOption(new Option('--sourceUrl', 'Remote url to use instead of local instance of Strapi'))
+  // .addOption(new Option('--sourceToken', 'Auth token for remote Strapi')) // required if sourceUrl is set
+  // .addOption(new Option('--archive', 'combine into one gzip file', true)) // for now we REQUIRE this to be true
+  // .addOption(
+  //   new Option(
+  //     '--only <data,to,include>',
+  //     'Comma-separated list of data to include (webhooks, content, localmedia, providermedia, config)', // ['webhooks', 'content', 'localmedia', 'providermedia', 'relations']
+  //     listOption
+  //   )
+  // )
+  // .addOption(
+  //   new Option(
+  //     '--exclude <data,to,exclude>',
+  //     'Comma-separated list of data to exclude (webhooks,content,localmedia,providermedia,config,relations)', // ['webhooks', 'content', 'localmedia', 'providermedia', 'relations']
+  //     listOption
+  //   )
+  // )
+  // .addOption(
+  //   new Option('--encryptionCipher <crypto cipher>', 'node crypto cipher to use', 'aes-256') // .choices(crypto.getCiphers())
+  // )
+  // .addOption(
+  //   new Option('--encryptionHash <crypto hash>', 'node crypto hash to use', 'sha-256') // .choices(crypto.getHashes())
+  // )
+  // .addOption(
+  //   new Option('--split <max MB per file>', 'split exported file when exceeding max filesize in MB')
+  // )
+  .hook('preAction', async (thisCommand) => {
+    const opts = thisCommand.opts();
+
+    // if encrypt is set but we have no key, prompt for it
+    if (opts.encrypt && !opts.key) {
+      try {
+        const answers = await inquirer.prompt([
+          {
+            type: 'password',
+            message: 'Please enter an encryption key',
+            name: 'key',
+            validate(key) {
+              if (key.length > 0) return true;
```

Comment:
Should we ask for a minimum length?

Reply:
Since this is just a user's local file and the security is up to them, I wasn't going to enforce anything if they want to use a weak password.